### PR TITLE
Create concourse pipeline & IAM roles in scripts

### PR DESCRIPTION
**Pull request by tlwr**
What
----

Creates a Concourse pipeline which mirrors repos to the GOV.UK Tools AWS account.

How to review
----

- Code review
- Run the `mirror-repos` job in [CD](https://cd.gds-reliability.engineering/teams/govuk-tools/pipelines/operations)

Ramifications
----

This may break the Git DR mirroring until it is switched over to the tools account

Todo
----

- [x] Slack notifications on failure

Further work
----

Move the pipeline from this repo into a more natural place for an `operations` pipeline

### CODE REVIEW

File: mirror_repos [hunk 1]
```diff
@@ -22,13 +38,30 @@ for repo in ${repos}; do
...
+    ${git} reflog expire --expire-unreachable=now --all
+    ${git} gc            --prune=now
+
+    echo "Pushing ${repo} --tags --force"
+    ${git} push --force --quiet --tags "${aws_remote}"
+
+    echo "Pushing ${repo} --all --force"
+    ${git} push --force --quiet --all "${aws_remote}"
+
+    echo "Manualling pushing ${repo}"
```
Comment (rubenarakelyan) [inline mirror_repos]: Manually :)

File: mirror_repos [hunk 2]
```diff
@@ -1,7 +1,21 @@
 #! /bin/bash
 
+set -ueo pipefail
+
+export AWS_REGION=eu-west-1
```
Comment (rubenarakelyan) [inline mirror_repos]: Can we do this in `eu-west-2` to be consistent with other stuff in the tools account?

**Review by rubenarakelyan — CHANGES_REQUESTED** [review]

File: mirror_repos (hunk 2 above)
Comment (tlwr) [reply mirror_repos]: Done

File: mirror_repos (hunk 1 above)
Comment (tlwr) [reply mirror_repos]: Done

**Review by rubenarakelyan — APPROVED** [review]

**Review by rubenarakelyan — APPROVED** [review]

**Pull request merged**